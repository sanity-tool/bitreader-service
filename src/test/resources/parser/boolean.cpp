static bool BR, B0, B1;

static void testBoolLogical() {
    BR = B0 && B1;
    BR = B0 || B1;
}