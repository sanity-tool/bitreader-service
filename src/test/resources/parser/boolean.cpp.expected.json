{
  "functions" : [ {
    "ref" : {
      "kind" : "Global",
      "index" : 0
    },
    "blocks" : [ {
      "instructions" : [ {
        "kind" : "LLVMLoad",
        "typeId" : 3,
        "operands" : [ {
          "kind" : "Global",
          "index" : 2
        } ],
        "sourceRef" : 0
      }, {
        "kind" : "LLVMTrunc",
        "typeId" : 5,
        "operands" : [ {
          "kind" : "Instruction",
          "blockIndex" : 0,
          "index" : 0
        } ],
        "sourceRef" : 0
      }, {
        "kind" : "LLVMBr",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Instruction",
          "blockIndex" : 0,
          "index" : 1
        }, {
          "kind" : "Block",
          "index" : 2
        }, {
          "kind" : "Block",
          "index" : 1
        } ],
        "sourceRef" : 0
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMLoad",
        "typeId" : 3,
        "operands" : [ {
          "kind" : "Global",
          "index" : 3
        } ],
        "sourceRef" : 0
      }, {
        "kind" : "LLVMTrunc",
        "typeId" : 5,
        "operands" : [ {
          "kind" : "Instruction",
          "blockIndex" : 1,
          "index" : 0
        } ],
        "sourceRef" : 0
      }, {
        "kind" : "LLVMBr",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Block",
          "index" : 2
        } ]
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMPHI",
        "typeId" : 5,
        "operands" : [ {
          "kind" : "Global",
          "index" : 4
        }, {
          "kind" : "Instruction",
          "blockIndex" : 1,
          "index" : 1
        } ]
      }, {
        "kind" : "LLVMZExt",
        "typeId" : 3,
        "operands" : [ {
          "kind" : "Instruction",
          "blockIndex" : 2,
          "index" : 0
        } ],
        "sourceRef" : 0
      }, {
        "kind" : "LLVMStore",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Instruction",
          "blockIndex" : 2,
          "index" : 1
        }, {
          "kind" : "Global",
          "index" : 5
        } ],
        "sourceRef" : 0
      }, {
        "kind" : "LLVMLoad",
        "typeId" : 3,
        "operands" : [ {
          "kind" : "Global",
          "index" : 2
        } ],
        "sourceRef" : 1
      }, {
        "kind" : "LLVMTrunc",
        "typeId" : 5,
        "operands" : [ {
          "kind" : "Instruction",
          "blockIndex" : 2,
          "index" : 3
        } ],
        "sourceRef" : 1
      }, {
        "kind" : "LLVMBr",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Instruction",
          "blockIndex" : 2,
          "index" : 4
        }, {
          "kind" : "Block",
          "index" : 3
        }, {
          "kind" : "Block",
          "index" : 4
        } ],
        "sourceRef" : 1
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMLoad",
        "typeId" : 3,
        "operands" : [ {
          "kind" : "Global",
          "index" : 3
        } ],
        "sourceRef" : 1
      }, {
        "kind" : "LLVMTrunc",
        "typeId" : 5,
        "operands" : [ {
          "kind" : "Instruction",
          "blockIndex" : 3,
          "index" : 0
        } ],
        "sourceRef" : 1
      }, {
        "kind" : "LLVMBr",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Block",
          "index" : 4
        } ],
        "sourceRef" : 1
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMPHI",
        "typeId" : 5,
        "operands" : [ {
          "kind" : "Global",
          "index" : 6
        }, {
          "kind" : "Instruction",
          "blockIndex" : 3,
          "index" : 1
        } ]
      }, {
        "kind" : "LLVMZExt",
        "typeId" : 3,
        "operands" : [ {
          "kind" : "Instruction",
          "blockIndex" : 4,
          "index" : 0
        } ],
        "sourceRef" : 1
      }, {
        "kind" : "LLVMStore",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Instruction",
          "blockIndex" : 4,
          "index" : 1
        }, {
          "kind" : "Global",
          "index" : 5
        } ],
        "sourceRef" : 1
      }, {
        "kind" : "LLVMRet",
        "typeId" : 2,
        "sourceRef" : 2
      } ]
    } ],
    "entryBlockIndex" : 0
  } ],
  "types" : [ {
    "kind" : "LLVMPointerTypeKind",
    "types" : [ 1 ]
  }, {
    "kind" : "LLVMFunctionTypeKind",
    "types" : [ 2 ]
  }, {
    "kind" : "LLVMVoidTypeKind"
  }, {
    "kind" : "LLVMIntegerTypeKind",
    "len" : 8
  }, {
    "kind" : "LLVMPointerTypeKind",
    "types" : [ 3 ]
  }, {
    "kind" : "LLVMIntegerTypeKind",
    "len" : 1
  } ],
  "globals" : [ {
    "kind" : "LLVMFunctionValueKind",
    "name" : "_ZL15testBoolLogicalv",
    "typeId" : 0
  }, {
    "kind" : "LLVMConstantIntValueKind",
    "typeId" : 3,
    "intValue" : 0
  }, {
    "kind" : "LLVMGlobalVariableValueKind",
    "name" : "_ZL2B0",
    "typeId" : 4,
    "operands" : [ {
      "kind" : "Global",
      "index" : 1
    } ]
  }, {
    "kind" : "LLVMGlobalVariableValueKind",
    "name" : "_ZL2B1",
    "typeId" : 4,
    "operands" : [ {
      "kind" : "Global",
      "index" : 1
    } ]
  }, {
    "kind" : "LLVMConstantIntValueKind",
    "typeId" : 5,
    "intValue" : 0
  }, {
    "kind" : "LLVMGlobalVariableValueKind",
    "name" : "_ZL2BR",
    "typeId" : 4,
    "operands" : [ {
      "kind" : "Global",
      "index" : 1
    } ]
  }, {
    "kind" : "LLVMConstantIntValueKind",
    "typeId" : 5,
    "intValue" : -1
  } ],
  "sourceRefs" : [ {
    "file" : "parser/boolean.cpp",
    "line" : 4
  }, {
    "file" : "parser/boolean.cpp",
    "line" : 5
  }, {
    "file" : "parser/boolean.cpp",
    "line" : 6
  } ]
}